src/Native/Bootstrap/main.cpp

```diff
@@ -328,23 +375,20 @@ int main(int argc, char* argv[])
     InitializeModules(nullptr, (void**)RtRHeaderWrapper(), 2, nullptr, 0);
 #endif // !CPPCODEGEN
 
-    int retval;
-#ifdef CPPCODEGEN
-    try
-#endif
-    {
-        retval = __managed__Main(argc, argv);
-    }
-#ifdef CPPCODEGEN
-    catch (const char* &e)
-    {
-        printf("Call to an unimplemented runtime method; execution cannot continue.\n");
-        printf("Method: %s\n", e);
-        retval = -1;
-    }
-#endif
-
-    RhpShutdown();
+#ifdef CORERT_DLL
+    // Run startup method immediately for a native library
+    __managed__Startup();
+#endif // CORERT_DLL
 
-    return retval;
+    return 0;
 }
+
+#ifdef CORERT_DLL
+static struct InitializeRuntimePointerHelper
```

Comment:
Does this work with optimizations on? I would think there needs to be something (e.g. linker directive on Windows - `pragma(comment(linker,"/include...` mentioned in one of the articles I linked earlier) to keep the static constructor alive.